libcxx/test/libcxx/utilities/memory/util.smartptr/util.smartptr.shared/libcxx.control_block_layout.pass.cpp

```diff
@@ -123,7 +123,9 @@ void test() {
 
   // Make sure both types have the same triviality (that has ABI impact since
   // it determined how objects are passed). Both should be non-trivial.
-  static_assert(std::is_trivial<New>::value == std::is_trivial<Old>::value, "");
+  static_assert((std::is_trivially_copyable<New>::value && std::is_trivially_default_constructible<New>::value) ==
+                    (std::is_trivially_copyable<Old>::value && std::is_trivially_default_constructible<Old>::value),
+                "");
```

Comment:
We can probably enforce that `is_trivially_copyable<New> == is_trivially_copable<Old> && ...`.